backend/cli_commands/SyncCommand.py

```diff
@@ -87,6 +85,63 @@ def sync_files(mission_path, mission):
                         )
                     except Exception as e:
                         logging.error(f"Error processing {mcap_path}: {e}")
+                else:
+                    file = File.objects.get(file=mcap_path)
+                # Get all video files in this folder
+                videos_in_folder = {
+                    video: os.path.join(subfolder_path, video)
+                    for video in storage.listdir(subfolder_path)[1]
+                    if video.endswith(".mp4")
+                }
+                logging.info(
+                    f"Found {len(videos_in_folder)} video files in folder {subfolder_path}"
+                )
+                # Process each topic in the metadata
+                for topic_name, topic_data in metadata.items():
+                    # Try to find a corresponding video file
+                    matching_video = None
+
+                    for video_filename in videos_in_folder.keys():
+                        # Match by topic name (adjust this logic if needed)
+                        if (
+                            topic_name.replace("/", "-") in video_filename
+                        ):  # Handle path-based topic names
+                            matching_video = videos_in_folder[video_filename]
+                            break
+
+                    if not matching_video:
+                        continue  # Skip this topic if no matching video is found
+
+                    # Ensure the video file exists in storage
+                    video_path = Path.joinpath(BASE_DIR, matching_video)
+                    if not storage.exists(video_path):
+                        logging.error(f"Video {video_filename} not found in storage.")
+                        continue
+
+                    with storage.open(video_path, "rb") as f:
+                        try:
+                            # Create and save Topic **before** adding video
+                            topic = Topic(
+                                file=file,
+                                name=topic_data["name"],
+                                type=topic_data["type"],
+                                message_count=topic_data["message_count"],
+                                frequency=topic_data["frequency"],
+                            )
+                            topic.full_clean()
+                            topic.save()
+
+                            # Save the video file to the topic
+                            topic.video.save(video_filename, ContentFile(f.read()))
+                            topic.save()
```

Comment:
```suggestion
                      try:
                          # Create and save Topic **before** adding video
                          topic = Topic(
                              file=file,
                              name=topic_data["name"],
                              type=topic_data["type"],
                              message_count=topic_data["message_count"],
                              frequency=topic_data["frequency"],
                          )
                          topic.full_clean()
                          topic.save()

                          # Save the video file to the topic
                          topic.video = matching_video
                          topic.save()
```
You don't need to read and write the video to add it to a topic, just assign the path.\
With this it is not possible add videos to topics that already exist in the database and topics that don't have a video don't get added.\
Maybe split the whole for loop into two for loops. One that adds all the topics and one that adds the videos to the topics.